### ISSUE TITLE
Feature request: improved support for native Xcode 8 tests, now that xctool won't be actively supported for it

### ISSUE THREAD

**Issue by benasher44** [travis-ci/travis-ci#6430]
More info on xctool support for Xcode 8: facebook/xctool#704.

tl;dr it seems to be that it won't be actively maintained outside of those in the community that still want to support it (although with enough community support, maybe this qualifies as active maintenance). My take is that this is probably because `xcodebuild` has improved a lot in Xcode 8 in terms of being a feature-full test runner:

- Can specifically include/exclude tests that you want `xcodebuild` to run
- 2 new build actions: `build-for-testing` and `test-without-building` (good for allowing one machine to build, while other machines just get the test build artifacts and run tests with them)

Specific things that would be nice to see supported:
- Specifying the `xcode_project`, `xcode_workspace`, and `xcode_scheme` [configuration keys](https://docs.travis-ci.com/user/languages/objective-c#Default-Test-Script) uses `xcodebuild` to run tests instead of `xctool` on Xcode 8 machines
- New configuration keys in `.travis.yml` for specifying the arguments to pass to the test runner to include or exclude specific tests (new in Xcode 8)

From the Travis infra side of things, it seems like the new `build-for-testing` and `test-without-building` actions would be pretty exciting because it could mean reduced time spent rebuilding the same test files when the goal is just to parallelize tests across multiple machines. The combination of new configuration keys to specify which tests to run + the Travis build matrix feature could make this intent easy to communicate in the `.travis.yml`.

**Comment by LavrL** [comment]
@benasher44  Do you have any updates how to run xctool for xcode 8?

**Comment by benasher44** [comment]
I just don't run it for Xcode 8. I use `xcodebuild`

**Comment by LavrL** [comment]
I use xctool 0.2.8 and OClint -0.10.1-x86-64-darwin-15.0.0. When I started to analyze code for xcode 8, it starts to fail.

How I do it :

I copy run-sonar.sh and sonar-project.properties in workspace of project and run run-sonar.sh. 

Before some days it worked perfectly.

Is it any other way to make it work?

**Comment by benasher44** [comment]
I'm not sure. I recommend asking in the xctool project.

**Comment by tamastimar** [comment]
If `xcodebuild` will be the default command please consider `xcpretty` also.
